src/Legend/Legend.tsx

```diff
@@ -96,6 +121,36 @@ export class Legend extends React.Component<LegendProps, LegendState> {
     if (extraParams && !(_isEqual(extraParams, prevProps.extraParams))) {
       this.setState({
         legendUrl: this.getLegendUrl(layer, extraParams)
+      }, this.setLegendSrc);
+    }
+  }
+
+  async setLegendSrc() {
+    const {
+      headers
+    } = this.props;
+
+    try {
+      this.setState({
+        loading: true
+      });
+
+      const response = await fetch(this.state.legendUrl, {
+        headers
+      });
+
+      if (!response.ok) {
+        throw new Error('No successful response returned while getting the legend graphic');
+      }
+
+      this.setState({
+        imgSrc: URL.createObjectURL(await response.blob())
```

Comment:
This should be revoked somewhere, although I don't know where this should be done.

Reply:
You're so right, thanks! Fixed in 00ca937c4091c7e6f4e30be3c59ac9e3609316ac.